### PR TITLE
update Score-P easyblock to use --with-nocross-compiler-suite=nvhpc for recent software versions + unset `$CPPFLAGS`, `$LDFLAGS`, `$LIB` which may interfere with configure magic

### PR DESCRIPTION
**Pull request by geimer**

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Score-P-7.0-gompi-2021a.eb_ 
 * **SUCCESS** _Score-P-8.0-gompi-2022a.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
jsczen2c1.int.jsc-zen2.easybuild-test.cluster - Linux Rocky Linux 8.5, x86_64, AMD EPYC 7742 64-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/SebastianAchilles/ca20e7a26347f7bc7df0a6800c79df96 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @SebastianAchilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Score-P-6.0-gompi-2019b.eb_ 
 * **SUCCESS** _Score-P-6.0-gompi-2020a.eb_ 
 * **SUCCESS** _Score-P-6.0-gompic-2019a.eb_ 
 * **SUCCESS** _Score-P-6.0-gompic-2019b.eb_ 
 * **SUCCESS** _Score-P-6.0-gompic-2020a.eb_ 
 * **SUCCESS** _Score-P-7.0-gompi-2020b.eb_ 
 * **SUCCESS** _Score-P-7.0-gompi-2021a.eb_ 
 * **SUCCESS** _Score-P-7.0-gompic-2020b.eb_ 
 * **SUCCESS** _Score-P-7.1-gompi-2021a-CUDA-11.3.1.eb_ 
 * **SUCCESS** _Score-P-8.0-gompi-2021b-CUDA-11.4.1.eb_ 
 * **SUCCESS** _Score-P-8.0-gompi-2021b.eb_ 
 * **SUCCESS** _Score-P-8.0-gompi-2022a-CUDA-11.7.0.eb_ 
 * **SUCCESS** _Score-P-8.0-gompi-2022a.eb_ 

Build succeeded for 13 out of 13 (13 easyconfigs in total)
cnx2 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8
See https://gist.github.com/SebastianAchilles/662f5558a4644992c03b14f319693d9d for a full test report.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/s/score_p.py [hunk 1]
```diff
@@ -56,16 +63,27 @@ def configure_step(self, *args, **kwargs):
...
                 toolchain.INTELCOMP: 'intel',
-                toolchain.NVHPC: 'pgi',
+                toolchain.NVHPC: 'nvhpc',
                 toolchain.PGI: 'pgi',
             }
+            nvhpc_since = {
+                'Score-P': '8.0',
+                'Scalasca': '2.6.1',
+                'OTF2': '3.0.2',
+                'CubeW': '4.8',
```
Comment (boegel) [inline easybuild/easyblocks/s/score_p.py:80]: @geimer Should `CubeW` here be `CubeWriter`?

Comment (boegel) [inline easybuild/easyblocks/s/score_p.py:80]: @geimer Should `CubeW` here be `CubeWriter`?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/s/score_p.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/s/score_p.py:80]: @SebastianAchilles Any thoughts on this?

Comment (geimer) [reply easybuild/easyblocks/s/score_p.py:80]: > @geimer Should `CubeW` here be `CubeWriter`?

Yes, indeed...  Will fix.

**Review by boegel — APPROVED** [review]

**Pull request merged**